### PR TITLE
Adding YatimaStdLib as a dependency

### PR DESCRIPTION
**Pull request by DanielRrr**
**Why**
Megaparsec has too many definitions which are irrelevant to parsing, they have a helping role

**What**
Solutions: All those definitions are moved to YatimaStdLib and imported. Such modules as RWST.lean and NEList.lean are removed. Now they're modules in YatimaStdLib.

### CODE REVIEW

File: Megaparsec/Errors/Bundle.lean [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 namespace Bundle
 
 structure ParseErrorBundle [stream : Stream.Stream S] {E : Type} where
-  errors : NEList.NEList (@StreamErrors.ParseError S E stream)
+  errors : NEList (@StreamErrors.ParseError S E stream)
   posState : ParserState.PosState S
 
 def toBundle {S : Type} [stream : Stream.Stream S] {E : Type}
-             (s : ParserState.State S E) (errs : NEList.NEList (@StreamErrors.ParseError S E stream))
+             (s : ParserState.State S E) (errs : NEList (@StreamErrors.ParseError S E stream))
```
Comment (cognivore) [inline Megaparsec/Errors/Bundle.lean:13]: Perhaps we want to show that we can close this now: https://github.com/yatima-inc/Megaparsec.lean/issues/18 ?

File: Megaparsec/MonadParsec.lean [hunk 2]
```diff
@@ -168,8 +168,8 @@ instance (E S : Type) [m : Monad M] [stream : Stream.Stream S]
             cok consumed s₁ []
         else
           let oops := match stream.chunkToTokens consumed with
-          | List.cons x xs => Errors.ErrorItem.tokens $ NEList.toNEList x xs
-          | [] => Errors.ErrorItem.label $ NEList.toNEList ' ' "Nothing consumed.".data
+          | List.cons x xs => Errors.ErrorItem.tokens $ List.toNEList x xs
```
Comment (cognivore) [inline Megaparsec/MonadParsec.lean:171]: 👏

File: Megaparsec/MonadParsec.lean [hunk 3]
```diff
@@ -230,9 +230,9 @@ instance (E S : Type) [m: Monad M] [a: Alternative M]
   label A f st := (mₚ.label E S (A × σ) f) ∘ st
   attempt A st := mₚ.attempt E S (A × σ) ∘ st
   lookAhead A state x := mₚ.lookAhead E S (A × σ) (state x)
-  notFollowedBy A state x := Util.seqComp (mₚ.notFollowedBy E S A (Util.fst <$> state x)) (pure (Unit.unit , x))
+  notFollowedBy A state x := Monad.seqComp (mₚ.notFollowedBy E S A ((fun (x,y) => x) <$> state x)) (pure (Unit.unit , x))
```
Comment (cognivore) [inline Megaparsec/MonadParsec.lean:233]: Isn't it just a `.1` projection?

**Review by cognivore — APPROVED** [review]
LGTM! The only change I didn't review was the `Either.fixs` one, because I forgot what's `fixs` and was too lazy to look it up.

File: Megaparsec/Errors/Bundle.lean (hunk 1 above)
Comment (DanielRrr) [reply Megaparsec/Errors/Bundle.lean:13]: What exactly?

**Pull request merged**